src/test/java/org/folio/rest/impl/ValidatorRegistryTest.java

```diff
@@ -463,14 +463,14 @@ public void shouldReturnBadRequestOnPutWhenNoImplementationReferenceSpecifiedFor
   }
 
   @Test
-  public void shouldReturnNotFoundWhenRuleDoesNotExist(final TestContext context) {
+  public void shouldReturnServerErrorWhenRuleDoesNotExist(final TestContext context) {
     requestSpecification()
       .header(TENANT_HEADER)
       .body(REGEXP_RULE_ENABLED.toString())
       .when()
       .put(TENANT_RULES_PATH)
       .then()
-      .statusCode(HttpStatus.SC_NOT_FOUND);
+      .statusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
We need several tests and correct status codes:
* If the rule neither contains "ruleId" nor "id" then it is a bad request (400).
* If the rule contains both "ruleId" and "id" and they differ then it is a bad request (400).
* If the record is not found it is 404 not found.

Can you also add tests similar to shouldUpdateExistingRule so that all three succeeding cases are covered ("ruleId" only, "id" only, "ruleId" == "id")?